src/main/test/com/epam/esm/dao/GiftCertificateDaoTest.java

```diff
@@ -0,0 +1,107 @@
+package com.epam.esm.dao;
+
+import com.epam.esm.config.SpringConfig;
+import com.epam.esm.entity.GiftCertificate;
+import com.epam.esm.mapper.GiftCertificateMapper;
+import com.epam.esm.utils.QueryConstructor;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.test.annotation.Rollback;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import java.math.BigDecimal;
+import java.time.Duration;
+import java.util.List;
+import java.util.Optional;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@ContextConfiguration(classes = SpringConfig.class)
+@ActiveProfiles("dev")
+public class GiftCertificateDaoTest {
+
+    private QueryConstructor constructor;
+    private GiftCertificateMapper mapper;
+    private GiftCertificateDao dao;
+    private GiftCertificate giftCertificate;
+    @Autowired
+    private JdbcTemplate template;
+
+    @Before
+    public void init() {
+        constructor = new QueryConstructor();
+        mapper = new GiftCertificateMapper();
+        giftCertificate = new GiftCertificate();
+        dao = new GiftCertificateDao(template, mapper, constructor);
+    }
+
+    @Test
+    public void testShouldFindById() {
+        //given
+        long id = 2;
+
+        //when
+        Optional<GiftCertificate> certificate = dao.findById(id);
+
+        //then
+        Assert.assertNotNull(certificate.get());
+    }
+
+    @Test
+    @Rollback
+    public void testShouldUpdateCertificates() {
+        //given
+        long id = 2;
+        giftCertificate.setName("test");
+        giftCertificate.setPrice(new BigDecimal(120));
+        giftCertificate.setDuration(Duration.ofDays(20));
+
+        //when
+        dao.updateGiftCertificate(2, giftCertificate);
+
+        //then
+        verify(dao, times(1)).updateGiftCertificate(id, giftCertificate);
```

Comment:
In this type of integration tests you shouldn't use `verify`, it's not a mock or spy to verify smth. You need to check the actual data in the DB.

Reply:
verify removed